models/comments.models.js

```diff
@@ -12,4 +12,22 @@ exports.selectCommentsByArticleId = (article_id) => {
             })
         }
     })
-}
\ No newline at end of file
+}
+
+exports.insertComment = (article_id, author, body) => {
+    if(!author || !body) {return Promise.reject ({status: 400, msg: "Bad Request"})}
+    return db.query('SELECT * FROM articles WHERE article_id = $1;', [article_id])
+    .then(({rows}) => {
+        if(rows.length === 0) {return Promise.reject({status: 404, msg: "Not Found"})}
+        else {return db
+            .query('INSERT INTO comments (author, body, article_id) VALUES ($1, $2, $3) RETURNING author, body;', [author, body, article_id])
+            .then(({rows}) => {
```

Comment:
This is nested  - make sure you move this out as we shouldn't have `then` blocks within other `then` blocks (I recommend closing it off between the `.query` and `.then` and then working from there ) :)